### PR TITLE
Allow to send service question if user is not logged in

### PR DESCRIPTION
**Pull request by goreck888**
Add a possibility to send question about the service if user is not logged in.
Disabled possibility to send a question if service contact emails are empty.

### CODE REVIEW

File: spec/features/service_spec.rb [hunk 1]
```diff
@@ -61,9 +69,23 @@
         fill_in("service_question_text", with: "text")
       end
 
-      expect { click_on "SEND" }.
+      expect { click_on "SEND"
+               sleep(5) }.
```
Comment (martaswiatkowska) [inline spec/features/service_spec.rb]: Are you sure it's necessary?

Comment (goreck888) [reply spec/features/service_spec.rb]: Yes, it don't wait for ajax response and this test fails without sleep.

File: config/locales/simple_form.en.yml [hunk 2]
```diff
@@ -31,8 +33,8 @@ en:
     hints:
       question:
         text: |
-          We will send and information service provider.
-          You will receive answer to %{email} email.
+          We will send your question to the Service provider.
+          You will receive an answer to %{email} email adress.
```
Comment (kmarszalek) [inline config/locales/simple_form.en.yml]: address

Comment (goreck888) [reply config/locales/simple_form.en.yml]: Yes. Sir!

**Comment by abacz** [comment]
@jarekzet please improve this modal.

prev: https://invis.io/CDU7THKRSKF#/386994593_Item-Detail_Modal_Question
zeplin: zpl.io/aneP7Ak

close btn & cloud
[modal_gfx.zip](https://github.com/cyfronet-fid/marketplace/files/3686319/modal_gfx.zip)

File: app/views/services/question_modal.js.erb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+$("#question-modal-title").html("<%= title %>")
```
[the remark this answers is not in the document]
Comment (goreck888) [reply app/views/services/question_modal.js.erb]: Changing $("#question-modal-body").html("<%= j render(form, service: @service, question: @question) %>") to document.getElementById("question-modal-body").innerHTML = ("<%= j render(form, service: @service, question: @question) %>") break reloading recaptcha.

[the remark this answers is not in the document]
Comment (martaswiatkowska) [reply app/views/services/question_modal.js.erb]: Why?

**Comment by goreck888** [comment]
@agpul @MinLidia E-mail content is updated now.

File: app/views/services/question_modal.js.erb (hunk 3 above)
[the remark this answers is not in the document]
Comment (goreck888) [reply app/views/services/question_modal.js.erb]: As I found out, innerHTML doesn't execute scripts, as jQuery .html()

**Comment by goreck888** [comment]
> Fix email content:
> 
> The question refers to the Service #Service:0x000055ba6d751458

Done.

File: app/views/services/question_modal.js.erb (hunk 3 above)
[the remark this answers is not in the document]
Comment (goreck888) [reply app/views/services/question_modal.js.erb]: Done, i don't think it's nice solution, but it works.

File: app/controllers/services/questions_controller.rb [hunk 4]
```diff
@@ -1,14 +1,47 @@
...
+
   def create
     @service = Service.friendly.find(params[:service_id])
-    @service.contact_emails.each  do |email|
-      ServiceMailer.new_question(email, current_user,
-                                 params[:service_question], @service).deliver_now
+    user = current_user || User.new(first_name: params[:service_question][:author].partition(" ").first,
+                                    last_name: params[:service_question][:author].partition(" ").last,
+                                    email: params[:service_question][:email])
+    @question = Service::Question.new(author: user.full_name,
```
Comment (mkasztelnik) [inline app/controllers/services/questions_controller.rb]: You can simplify this a bit:

```ruby
 @question = Service::Question.new(author: user&.full_name || params[:service_question][:author],
                                                            email: user&.email || params[:service_question][:email], ...)
```
Then you will not need to create fake user object.

File: app/controllers/services/questions_controller.rb [hunk 5]
```diff
@@ -1,14 +1,47 @@
...
+                                     email: user.email,
+                                     text: params[:service_question][:text],
+                                     service: @service)
+    respond_to do |format|
+      if @question.valid? && verify_recaptcha
+        @service.contact_emails.each  do |email|
+          ServiceMailer.new_question(email, @question, @service).deliver_now
```
Comment (mkasztelnik) [inline app/controllers/services/questions_controller.rb]: It is better to send emails in a separate thread. To do this use the `deliver_later` method.

File: app/controllers/services/questions_controller.rb (hunk 4 above)
Comment (goreck888) [reply app/controllers/services/questions_controller.rb]: Done.

File: app/controllers/services/questions_controller.rb (hunk 5 above)
Comment (goreck888) [reply app/controllers/services/questions_controller.rb]: Done.

**Pull request merged**